### ISSUE TITLE
`start ignoring` ignores too much

### ISSUE THREAD

**Issue by adtac** [coala/coala#2898]
```
test=4
# Start ignoring *
test2=5
# Stop ignoring
test3=6
```

Running PEP8Bear on this doesn't produce any results for some reason (it should flag lines 1 and 5). However this works properly:

```
test=4

# Start ignoring *
test2=5
# Stop ignoring

test3=6
```

**Comment by jarifibrahim** [comment]
Moved to https://github.com/coala/coala-bears/issues/1026

**Comment by adtac** [comment]
@RohanVB unassigned because of inactivity

**Comment by sils** [comment]
that's not a bug in split_diff, it's intended behaviour, not sure we can do something to fix this issue but don't just change split diff @jarifibrahim , that's dangerous  and affects a lot of bears!

**Comment by jarifibrahim** [comment]
@sils If we don't change the `split_diff` method then every bear has to be modified. This issue exists with all the bears that use this method. I tried `SpaceConsistencyBear` and found similar issue. Either we change the default value or we'll have to explicitly change the bears that call the `split_diff` method.

**Comment by sils** [comment]
The SPaceConsistencyBear doesn't use split_diff

**Comment by jarifibrahim** [comment]
I meant `PyUnusedCodeBear`. It uses `split_diff`

**Comment by jarifibrahim** [comment]
Should we update following the bears?

```
bears/general/IndentationBear.py:
bears/general/coalaBear.py:
bears/js/JSONFormatBear.py:
bears/matlab/MatlabIndentationBear.py:
bears/python/PEP8Bear.py:
bears/python/PEP8NotebookBear.py: 
bears/python/PyCommentedCodeBear.py:
bears/python/PyUnusedCodeBear.py
bears/python/YapfBear.py:    
```
All these bears use `split_diff` with default `distance` value.

**Comment by sils** [comment]
I'm saying we shouldn't change the distance value carelessly. It's there for a reason and diffs will be fragmented and might not make sense when reducing the min distance.

**Comment by Makman2** [comment]
The right solution would be to change all bears and extract diffs that really match a certain problem. For example:
```
#spacing error 1
#spacing error 2
```
would have ideally two results except of one. Problem is linter doesn't make a distinction, it just tries to group changes in the file reasonably, using the diff distance. Another problem is that most tools are not smart enough to provide diffs for single problems, either they just say "This is wrong" or they just give the complete corrected file.
We can make our bears only as smart as the underlying tools in this manner (except we run our own detection heuristic over the corrections, this could could work for simple spacing issues or similar, but in many other cases this would lead to an overkill, resulting in a reimplementation of the tool itself).

To make ignores work again, it's possible to change the `diff_distance` in `@linter` to 1 when `output_format='corrected'`, but like @sils says, this leads to fragmentation, and it's very questionable as it varies on the underlying problem how good this approach is^^ On the other hand the issue already exists with our default distance of 2, though minimized.

So what to do?
I don't know for sure myself, ideally bears have to get smarter. I would consider our current approach of ignoring results is "ideal". A result with a certain range/diff means that everything inside it is responsible or even the cause of the issue, not just a part. For example you can't change a variable name (e.g. coala detects you've chosen a wrong name, whyever) without replacing all instances. Just replacing 3/4 makes the patch invalid^^

That's so far from my side.

PS
I think when we get coala-aspects ready, diff-quality will improve as we gonna start to pick out specific functionality from tools, or we are going to write stuff ourselves.

**Comment by jarifibrahim** [comment]
I'm not really sure what needs to be done. Could anyone please assign this issue to someone else?

**Comment by Makman2** [comment]
yeah this one is really hard regarding the concept... sorry that it has turned out like this^^

**Comment by adtac** [comment]
at least a difficulty/medium then?

**Comment by parvmor** [comment]
Hey! I think that the error is mainly in `/coalib/processes/Processing.py` as it is yielding the ranges to be ignored.

**Comment by parvmor** [comment]
Hey! I think the error is in `/coalib/processes/Processing.py` as it implements the functionality for ignoring the ranges.

**Comment by Makman2** [comment]
Yeah but it's not about the functionality itself, it's about the concept. Read the issue discussion :+1:
